openquake/hazardlib/gsim/nath_2012.py

```diff
@@ -0,0 +1,201 @@
+    #: this GSIM can calculate. A set should contain classes from
+    #: module :mod:`openquake.hazardlib.imt`.
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([SA, PGA, PGV])
+
+    #: In simulations only the vertical component is estimated (see p. 479)
+    #: and the stochastic dataset is what the GMPE is based on, so this model
+    #: effectively predicts vertical motions.
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.VERTICAL
+
+    #: The only sigma is reported in the main coefficient table, Table 5 on
+    #: p. 483, and must be the total standard deviation.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([const.StdDev.TOTAL])
+
+    #: Required site parameter Vs30 is used to determing the NEHRP
+    #: site class, and thus to choose site amplification coefficients
+    #: and site amplification stanard error from Table 5 on p. 208.
+    REQUIRES_SITES_PARAMETERS = set()
+
+    #: Sole required rupture parameter is magnitude, since faulting
+    #: style is not addressed.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: It is noted that "r_rup is the fault-rupture distance in kilometers"
+    #: following equation (11) on p. 484.
+    REQUIRES_DISTANCES = set(('rrup',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+
+        Implements equation (11) on p. 484:
+
+        ``ln(P) = c1 + c2*M + c3*(10 - M)^3 + c4*ln(R + c5*exp(c6*M)``
+        """
+
+        assert imt.__class__ in self.DEFINED_FOR_INTENSITY_MEASURE_TYPES
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+
+        # obtain coefficients for required intensity measure type (IMT)
+        coeffs = self.COEFFS_BEDROCK[imt].copy()
+
+        # obtain IMT-independent coefficients
+        coeffs.update(self.CONSTS)
+
+        # compute bedrock motion, equation (11)
+        ln_mean = self._compute_mean(rup, dists, coeffs)
+
+        # obtain standard deviation
+        ln_stddev = self._get_stddevs(coeffs)
+
+        return ln_mean, [ln_stddev]
+
+    def _compute_mean(self, rup, dists, coeffs):
+        """
+        Evaluate equation (11) on p. 484.
+        """
+
+        ln_p = coeffs['c1'] + coeffs['c2']*rup.mag + \
+            coeffs['c3']*(self.CONSTS['ref_mag'] - rup.mag)**3 +\
+            coeffs['c4']*np.log(dists.rrup +
+                                coeffs['c5']*np.exp(coeffs['c6']*rup.mag))
+
+        return ln_p
+
+    @classmethod
+    def _get_stddevs(cls, coeffs):
+        """
+        Look up from Table 5 on p. 483. Although the value is listed as
+        "sigma_log(Y)" in that table it is assumed that this is a natural
+        logarihm as for ln_p in equation (11) rather than the common logarithm.
+        """
+
```

Comment:
Good catch - you're right. Otherwise the residuals would be unbelievably small. Also this interpretation is consistent with the usage of log for log_10 and ln for log_e in the paper. I assumed, incorrectly, that the same logarithm used for the mean would be used for the stddev. I need to pay more attention to the orders of magnitudes of the tabulated sigmas.